src/extension.ts

```diff
@@ -420,14 +415,78 @@ export function activate(context: ExtensionContext) {
         xml['xml']['catalogs'].push(catalog);
       }
     })
-    externalXmlSettings.xmlFileAssociations.forEach(element => {
-      if (!xml['xml']['fileAssociations'].some(fileAssociation => fileAssociation.systemId === element.systemId)) {
-        xml['xml']['fileAssociations'].push(element);
-      }
-    });
+    const variableSubstitutedAssociations: XMLFileAssociation[] =
```

Comment:
Embed the following code in a  function like `getVariableSubstitutedAssociations(XMLFileAssociation[] associations)` that you move into variableSustition.ts